index.js

```diff
@@ -369,17 +369,24 @@ class ResembleHelper extends Helper {
     const helper = this._getHelper();
     await helper.waitForVisible(selector);
     const els = await helper._locate(selector);
-    if (!els.length) throw new Error(`Element ${selector} couldn't be located`);
-    const el = els[0];
+    
+    if (this.helpers['TestCafe']) {
+      if (await els.count != 1) throw new Error(`Element ${selector} couldn't be located or isn't unique on the page`);
+    }
+    else {
+      if (!els.length) throw new Error(`Element ${selector} couldn't be located`);
+    }
 
     let location, size;
 
     if (this.helpers['Puppeteer']) {
+      const el = els[0];
```

Comment:
what is the need for this?

Reply:
I moved this assignment from line 373 here. Because that doesn't work like this for TestCafe.